### ISSUE TITLE
Plot title (and signature, sources).

### ISSUE THREAD

**Issue by Fil** [observablehq/plot#92]
as part of legends

**Comment by ericemc3** [comment]
According to the usual rules of legibility, a chart should include an informative title, coming first (above the chart), possibly a subtitle, obligatorily a source, at the bottom of the chart, more optionally a copyright (which can also be arranged vertically, e.g. bottom right).

![image](https://user-images.githubusercontent.com/17407796/118007833-ce99bb80-b34c-11eb-9bf8-780cdab39b77.png)

ggplot2 and vega-lite allow you to specify these essential elements, it would be very nice if Plot could as well.

The SVG export of the Plot should allow you to get the whole graphic, with titles and sources.

As vega-lite proposes, specifying a title as an array allows it to be spread over several lines.

The height specified for the graphic should not take these elements into account, but only the central body of the graphic.

**Comment by breck7** [comment]
Just another vote in support of title and subtitle support :). I would take @martgnz 's demo (https://github.com/observablehq/plot/pull/423) right now as it looks good enough and would unblock me. This is the one thing making it hard to build *ready-to-share* charts in Plot.

At Our World in Data the titles and subtitles are key, and the charts would be shared thousands of times less without them.

![coronavirus-data-explorer](https://user-images.githubusercontent.com/74692/178867508-5f44e78b-bfed-4969-8d85-dc246a140723.png)

**Comment by kentr** [comment]
Have [`<title>` and `aria-labelledby`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title) been considered?

**Comment by Fil** [comment]
We haven't considered aria-labelledby for now, because it needs an associated element with a given id, which makes it a bit complicated to handle if we want to be able to have several charts on a single page (they're not supposed to share ids). But we'll definitely consider it if there is a clear use case. Not that this property is usually visible only to screen readers, so I seems to fall under a different category, and might need a separate issue?

**Comment by mbostock** [comment]
I think maybe @kentr’s comment is regarding tooltips—which we already support via the **title** option; also note that we also already support **ariaLabel** and **ariaDescription** options. This issue is primarily regarding a visual title element, i.e., a heading above the chart.

**Comment by kentr** [comment]
@mbostock @Fil 

My comment was regarding a title element for the whole SVG (not tooltips).

I envision markup like this:

```html
<div id="svg-title">Daily new confirmed COVID-19 deaths per million people</div>
  <svg aria-labelledby="svg-title" ...>
    <g>
      ...
    </g>
</svg>
```

Originally, I didn't know that this issue was regarding a _visual_ title, so I thought of `<title>` as a candidate for a non-visual title and `aria-labelledby` as a candidate for a visual title in an external element. 

I'm thinking of explicitly connecting the "title" discussed in this issue to the SVG, for accessibility. For accessible titles, when there is already an element on the page that has the title text, [MDN recommends](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title) connecting it to the SVG with `aria-labelledby`. 

Plot supports `aria-label`, but if there's already a title element on the page, why not use that?

> If an element can be described by visible text, it is recommended to reference that text with an [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attribute rather than using the `<title>` element.

But the [smashingmagazine.com article suggests](https://www.smashingmagazine.com/2021/05/accessible-svg-patterns-comparison/#basic-alternative-descriptions-using-the-svg-tag-group-2) that even with the `aria-labelledby` connection, adding a `<title>` element to the SVG is better for some browsers.

Using `aria-labelledby` does require a unique `id` attribute on the title element, but that seems doable. And having an `aria-labelledby` option lets the user provide their own value if they already have the title element on the page from something else (for example, when the plots are in a react app and there are already DOM elements with the SVG titles).

**Comment by maxlvhao** [comment]
frustrated by not able to add a proper title, i ended up cheating by using the following code
`    Plot.text(['Question: My university has a campus-wide generative AI (e.g. ChatGPT) policy for students.'], {frameAnchor: "Top",dy: -25}),`

within my Marks:

I had to add marginTop:50 to make sure my title doesn't get cut off..

<img width="950" alt="Screenshot 2023-07-21 at 8 38 53 AM" src="https://github.com/observablehq/plot/assets/12852655/e616e9f5-0a1b-4855-998d-17fd93676e7e">

**Comment by maxlvhao** [comment]
> frustrated by not able to add a proper title, i ended up cheating by using the following code ` Plot.text(['Question: My university has a campus-wide generative AI (e.g. ChatGPT) policy for students.'], {frameAnchor: "Top",dy: -25}),`
> 
> within my Marks:
> 
> I had to add marginTop:50 to make sure my title doesn't get cut off..
> 
> <img alt="Screenshot 2023-07-21 at 8 38 53 AM" width="950" src="https://user-images.githubusercontent.com/12852655/255174598-e616e9f5-0a1b-4855-998d-17fd93676e7e.png">

after some more tweaks..
<img width="768" alt="Screenshot 2023-07-21 at 8 47 49 AM" src="https://github.com/observablehq/plot/assets/12852655/3ff1805b-0718-4fbc-82de-8a2c2b20b70e">

`Plot.plot({
  marginLeft:180,
  marginTop:50,
  //caption: "Figure 2. Response to question 'My university has a campus-wide generative AI (e.g. ChatGPT) policy for students.' ",
  x:{grid: true, label: "Number of responses"},
  y:{label:null},
  marks: [
    Plot.barX(survey, Plot.groupY({x: "count"}, {y: "My university has a campus-wide generative AI (e.g. ChatGPT) policy for students.", sort: {y: "x"}})),
    Plot.text([`Question: My university has a campus-wide generative AI (e.g. ChatGPT) 
policy for students.`],{frameAnchor: "top-left", dy: -40, fontWeight: "bold", fontSize:"12", textAnchor:"start"}),
    Plot.ruleX([0])
  ]
})`
